packages/core/src/browser/preferences/preference-contribution.ts

```diff
@@ -135,6 +135,13 @@ export class PreferenceSchemaProvider extends PreferenceProvider {
     }
 
     protected doSetSchema(schema: PreferenceSchema): PreferenceProviderDataChange[] {
+        const ajv = new Ajv();
+        const valid = ajv.validateSchema(schema);
+        if (!valid) {
+            const errors = !!ajv.errors ? JSON.stringify(ajv.errors) : 'unknown validation error';
```

Comment:
Have you tried https://github.com/epoberezkin/ajv#errorstextarrayobject-errors--object-options---string? Does it produce better formatted error messages?

Reply:
will try! thanks!

Reply:
@akosyakov I updated it to print the error message like this:

```
An invalid preference schema was rejected: data.properties['rust-client.channel'].type should be equal to one of the allowed values, data.properties['rust-client.channel'].type[1] should be equal to one of the allowed values, data.properties['rust-client.channel'].type should match some schema in anyOf
```